internal/server/statvar_search.go

```diff
@@ -104,12 +104,33 @@ func searchTokens(
 	tokens []string, index *resource.SearchIndex) ([]*pb.EntityInfo, []*pb.EntityInfo) {
 	svCount := map[string]int{}
 	svgCount := map[string]int{}
+
+	// Get all matching sv and svg from the trie for each token
 	for _, token := range tokens {
-		for sv := range index.TokenSVMap[token] {
-			svCount[sv]++
+		currNode := index.RootTrieNode
```

Comment:
Can you add comments for each block of logic? Here is logic intensive code :)

Reply:
Also add some example case in the comment.

Reply:
done

Reply:
nice!